### ISSUE TITLE
Naming of OVE concepts such as space/section/clients

### ISSUE THREAD

**Issue by dsigdosystem** [ove/ove-docs#2]
<a href="https://github.com/jamesscottbrown"><img src="https://avatars1.githubusercontent.com/u/338833?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jamesscottbrown](https://github.com/jamesscottbrown)**
_Thursday Sep 13, 2018 at 12:17 GMT_
_Originally opened as https://github.com/dsi-icl/ove/issues/25_

----

Currently, several OVE ``clients`` form a ``space``, on which several ``sections`` can be created, each of which runs a single ``app``.

I initially found these terms slightly difficult to remember (particularly ``section``), so tried to think of some alternatives:

* Several OVE ``clients`` form a ``surface`` on which several ``panels`` can be created
* ~~Several OVE ``clients`` form a ``surface`` on which several ``windows`` can be created~~
* ~~Several OVE ``clients`` form a ``panel`` on which several ``windows`` can be created~~
* Several OVE ``clients`` form a ``board`` on which several ``panels`` can be created
* ~~Several OVE ``clients`` form a ``board`` on which several ``windows`` can be created~~
* ~~Several OVE ``clients`` form a ``display`` on which several ``windows`` can be created~~
* Several OVE ``clients`` form a ``display`` on which several ``panels`` can be created

Personally, I think that ``window`` is the clearest name: ``panel`` and ``section`` both seem more suggestive of non-overlapping parts. However, there is a possibility of confusion that an OVE window would not correspond to a browser window (i.e., a window on the whole display is different to a window on one of the monitors).

``display`` seems an appropriate name, as what is currently called a space is a (typically large, high-resolution) tiled display.

Do **you** think any of these are clearer than the current names?

**Comment by dsigdosystem** [comment]
<a href="https://github.com/senakafdo"><img src="https://avatars3.githubusercontent.com/u/3415383?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [senakafdo](https://github.com/senakafdo)**
_Thursday Sep 13, 2018 at 18:36 GMT_

----

I personal did not like the names I gave these different components. I had to call it something so I chose some names for it - as I also explained in a previous team meeting. I think some of the terminology used does seem to be accurate: client, app, layout. But some can be fixed: section, space. But, I'm not fully convinced (even after reading this issue, :)..) that we have better names for it.

The term `window` can be extremely confusing since the JS DOM also has a window object and so forth. Similarly, the term `display` is also somewhat confusing since people associate that with a single screen. So, I don't think either works.

For space - I don't personally find it hard to interpret. Some people call these `spaces`, some call them `workspaces`, and some call them `worlds`. If our documentation has a nice diagram explaining what we mean by `space` I don't expect any confusion to follow. However, I didn't exactly mean "several OVE clients form a space".

To see if anything works lets see if you can replace the use of these terms in the following paragraph:
OVE supports multiple `spaces`. Each space is associated with a set of `displays` which are generally (not necessarily) grouped together to form a large high resolution display environment. Within each `space` OVE is deployed as a collection of `clients` placed accordingly using a coordinate system. A `layout` configuration describes the placement of these clients. Users of OVE can create `sections` on this `layout` into which OVE `apps` are deployed. These sections can be overlapping, and can be a rectangle of any size upto a maximum that is limited by the dimensions of the `space` itself.

The word `section` may not be quite fitting but I also don't find that we can subsitute the word section with `display` or `window` without them confusing the reader even further. `partition` may work, but generally they are not thought to be overlapping. Other terms like `containers`, `groups`, `clusters` all mean different things, so I believe that we potentially have the most appropriate term.

**Comment by dsigdosystem** [comment]
<a href="https://github.com/jamesscottbrown"><img src="https://avatars1.githubusercontent.com/u/338833?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jamesscottbrown](https://github.com/jamesscottbrown)**
_Thursday Sep 13, 2018 at 19:20 GMT_

----

I'm also not fully convinced myself: I will continue trying to think of better terms.

I agree that ``window`` would be too overloaded. This reduces my suggested alternatives to:

* ``space`` -> ``board``, ``surface``, or ``display``
* ``section`` -> ``panel`` (or ``area``)

On further thought, ``panel`` has the disadvantage of potential confusion with an LCD display panel (i.e., a single monitor).

I'm not sure whether people really do associate ``display`` with a single screen/monitor: I think it is standard to refer to a grid of screens forming a single display surface as a ``display`` (as in "large high-resolution display"). 

I would therefore say that *"Each space is associated with a set of ``displays``"* is confusing: I think that we should avoid referring to the individual monitors as displays, regardless of whether ``space`` is renamed.

> Within each space OVE is deployed as a collection of ``clients`` placed accordingly using a coordinate system. A ``layout`` configuration describes the placement of these ``clients``. Users of OVE can create ``sections`` on this`` layout`` into which OVE ``apps`` are deployed.

I think it would be clearer if ``sections`` are created directly on a ``space`` (rather than on a ``layout``), and the ``layout`` defines how to map from ``space`` coordinates to coordinates on each ``client``.

**Comment by dsigdosystem** [comment]
<a href="https://github.com/senakafdo"><img src="https://avatars3.githubusercontent.com/u/3415383?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [senakafdo](https://github.com/senakafdo)**
_Thursday Sep 13, 2018 at 21:30 GMT_

----

 > I think it is standard to refer to a grid of screens forming a single display surface as a display (as in "large high-resolution display"

That's a good point. Having surveyed a number of such implementations I can confidentally say that refers to the combined screen surface area which is created by the amalgamation of a number of monitors/panels or projection surfaces.

> "Each space is associated with a set of displays" is confusing

As noted above, you are spot-on.

> I think it would be clearer if sections are created directly on a space (rather than on a layout), and the layout defines how to map from space coordinates to coordinates on each client.

Agreed. We should polish the way we introduce these technical terms to a lay person. One of the biggest weaknesses in comparable frameworks is that their terminology tends to very confusing. We should avoid getting into the same situation.

So, in summary - I see no reason to change what we have, so far.

**Comment by dsigdosystem** [comment]
<a href="https://github.com/senakafdo"><img src="https://avatars3.githubusercontent.com/u/3415383?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [senakafdo](https://github.com/senakafdo)**
_Monday Oct 01, 2018 at 20:06 GMT_

----

from #84, we also need to discuss in detail about `oveClientId` and its format `{space}-{client}.{sectionId}` which is described on [this comment](https://github.com/dsi-icl/ove/blob/0bf84290f5e9f51005bea63e17c5ac50ba37036f/packages/ove-core/src/client/ove.js#L121).

**Comment by senakafdo** [comment]
**Hardware Concepts:**
Clients.json -> Spaces.json
Space -> Space
Screen -> Screen
Machine -> Machine

**Software Concepts:**
ove.layout -> ove.geometry
Layout (Layout Service) -> Layout
Canvas -> oveCanvas
Container -> Group
Section -> Section
Geometry -> Section.Geometry
SectionId -> Section.Id

Client -> Client
Server -> Server

App -> App
Layer -> App.Layer (only used in Maps App)
Config -> App.Config
State -> State
View -> View
ClientId -> View.Id
Control -> Control

**Project/Data Concepts:**
Asset -> Asset
Visualisation -> Project
State.json -> Project.json

**Comment by senakafdo** [comment]
All naming is now fixed (in code and in docs) as discussed.
